icu4j/main/classes/core/src/com/ibm/icu/impl/RBBIDataWrapper.java

```diff
@@ -359,8 +378,8 @@ public static RBBIDataWrapper get(ByteBuffer bytes) throws IOException {
         }
         ICUBinary.skipBytes(bytes, This.fHeader.fRuleSource - pos);
         pos = This.fHeader.fRuleSource;
-        This.fRuleSource = ICUBinary.getString(
-                bytes, This.fHeader.fRuleSourceLen / 2, This.fHeader.fRuleSourceLen & 1);
+        This.fRuleSource = new String(
+            ICUBinary.getBytes(bytes, This.fHeader.fRuleSourceLen, 0), "UTF8");
```

Comment:
Doesn't the fRuleSourceLen include trailing NULs that should be omitted?
I might be wrong. Please check/confirm.

Reply:
No. I didn't include that in fRuleSourceLen